### ISSUE TITLE
Gulp serve terminates or is hanging when CSS Compilation fails

### ISSUE THREAD

**Issue by StfBauer** [SharePoint/sp-dev-docs#320]
#### Category
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Enhancement / Suggestion
- [ ] Additional article idea

#### Expected or Desired Behavior
Continue without restarting gulp serve

#### Observed Behavior
<img width="638" alt="screen shot 2016-12-15 at 23 25 47" src="https://cloud.githubusercontent.com/assets/5503835/21244477/1efc8ec8-c31e-11e6-9684-3658993b1535.png">
With this error even when it has been fixed a restart of `gulp serve` is required.

#### Steps to Reproduce
Place a 'test.css' or '_test.sass' file beneath the folder tests. Open main style sheet and add the following line.
@import 'test.css' or @import 'test' in the '*.module.css' and save. After the file have been moved to the location and the main sass file have been saved nothing happens anymore.

**Comment by chakkaradeep** [comment]
@iclanton may provide more information here but I think the build system requires SASS and not CSS. So, either, `test.scss` or `test.module.scss` should work. Can you try that?

**Comment by StfBauer** [comment]
It's not related to the file itself. The problem arises when I try to import a file that doesn't exist. Most of the times it's working but sometime I run in this problem.
It is also related to another problem that you cannot gulp serve when there is a problem in the sass compliation. In order to start I need to first fix the problem and then I'm able to run gulp serve. Would be better, if possible, to start anyway and fix the problem later.

I'll monitor this further and do verbose debugging in SPFx from now on to provide more informations on this issue.

**Comment by iclanton** [comment]
@nickpape-msft - is this part of the other SCSS work that you're doing? Can you take a look at this?

**Comment by nickpape-msft** [comment]
No, it sounds like the primary issue is that the SASS task failure is causing the watch task to stop watching and rebuilding. This is probably an actual bug, since "gulp serve" is considered to be an inner-loop scenario.

**Comment by nickpape-msft** [comment]
@StfBauer for the initial issue, I am unable to reproduce the problem. It is true that the compilation bails out after the SASS failure (with the last line being "Finished subtask 'copyStaticAssets'"), but as soon as I change the SASS file again compilation correctly restarts.

As for the second issue, it is true that if any of the build tasks fail during the initial compilation we never make it to the "serve" task. This was due to some quirky behavior around watch() and how we defined the tasks. I've remedied this in gulp-core-build, so we will see the fix in the next drop.
